src/payment/strategies/cybersource/cybersource-threedsecure-payment-processor.ts

```diff
@@ -0,0 +1,238 @@
+import { includes, some } from 'lodash';
+import { Subject } from 'rxjs/index';
+import { filter } from 'rxjs/internal/operators';
+import { take } from 'rxjs/operators';
+
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import {
+    MissingDataError,
+    MissingDataErrorType,
+    NotInitializedError,
+    NotInitializedErrorType,
+    RequestError,
+    StandardError
+} from '../../../common/error/errors';
+import { OrderActionCreator, OrderPaymentRequestBody, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { CreditCardInstrument, ThreeDSecure } from '../../payment';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethod from '../../payment-method';
+import { PaymentRequestOptions } from '../../payment-request-options';
+
+import {
+    CardinalEventAction,
+    CardinalEventResponse,
+    CardinalEventType,
+    CardinalInitializationType,
+    CardinalPaymentBrand,
+    CardinalPaymentStep,
+    CardinalTriggerEvents,
+    CardinalValidatedAction,
+    CardinalValidatedData,
+    CyberSourceCardinal,
+    CyberSourceScriptLoader,
+    SetupCompletedData,
+    SignatureValidationErrors,
+} from './index';
+
+export default class CyberSourceThreeDSecurePaymentProcessor {
+    private _Cardinal?: CyberSourceCardinal;
+    private _paymentMethod?: PaymentMethod;
+    private _cardinalEvent$: Subject<CardinalEventResponse>;
+    private _isSetupCompleted = false;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _cyberSourceScriptLoader: CyberSourceScriptLoader
+    ) {
+        this._cardinalEvent$ = new Subject();
+    }
+
+    initialize(paymentMethod: PaymentMethod): Promise<InternalCheckoutSelectors> {
+        if (this._isSetupCompleted) {
```

Comment:
I commented you guys about this validation, initialize this value always will be `false`, you change your selection of provider and come back to Cybersource you should call `deinitialize` method keep all values as initially are.